### PR TITLE
Discord Rich Presence options

### PR DESCRIPTION
**Pull request by rogeraabbccdd**
- Change Discord flag to NoRequireDiscord, here's more info from [documentation](https://discord.com/developers/docs/game-sdk/sdk-starter-guide#code-primer-unity-csharp)
  ```
    Discord.CreateFlags.Default will require Discord to be running for the game to work
    If Discord is not running, it will:
    1. Close your game
    2. Open Discord
    3. Attempt to re-open your game
    Step 3 will fail when running directly from the Unity editor
    Therefore, always keep Discord running during tests, or use Discord.CreateFlags.NoRequireDiscord
  ```
- Add Discord Rich Presence in options, this also adds 4 new language string
  ```
    options_discord_header
    options_discord_rich_presence,Enable
    options_discord_rich_presence_reconnect_label
    options_discord_rich_presence_reconnect_button
  ```
  TECHMANIA_UOCpDiHdvl

### CODE REVIEW

File: TECHMANIA/Assets/Scripts/DiscordController.cs [hunk 1]
```diff
@@ -23,8 +24,10 @@ public class DiscordController
 
     public static void Start ()
     {
-        if (discord != null || !SupportedOnCurrentPlatform()) return;
-        discord = new Discord.Discord(802017593086836767, (UInt64)Discord.CreateFlags.Default);
+        if (discord != null || !SupportedOnCurrentPlatform() || !Options.instance.discordRichPresence) return;
+        try {
+            discord = new Discord.Discord(802017593086836767, (UInt64)Discord.CreateFlags.NoRequireDiscord);
+        } catch {}
```
Comment (macmillan333) [inline TECHMANIA/Assets/Scripts/DiscordController.cs:30]: Do we know what exceptions `Discord.Discord` may throw? Can we catch and handle them specifically? Or at least catch `Exception` and print it in console?

Same with `Dispose()`.

Comment (rogeraabbccdd) [reply TECHMANIA/Assets/Scripts/DiscordController.cs:30]: `Discord.Discord` will throw this error if Discord is not running.
```
ResultException: InternalError
Discord.Discord..ctor (System.Int64 clientId, System.UInt64 flags) (at Assets/Plugins/DiscordGameSDK/Core.cs:1017)
DiscordController.Start () (at Assets/Scripts/DiscordController.cs:28)
Startup.Start () (at Assets/Scripts/Components/Main Menu/Startup.cs:18)
```

Comment (rogeraabbccdd) [reply TECHMANIA/Assets/Scripts/DiscordController.cs:30]: And looks like `Dispose()` does not throw any error message.
But `discord.RunCallbacks()` throws this error if Discord is not running, this happens if Discord is closed while playing the game.
I'm working on new commit to fix this.
```
ResultException: NotRunning
Discord.Discord.RunCallbacks () (at Assets/Plugins/DiscordGameSDK/Core.cs:1056)
DiscordController.RunCallbacks () (at Assets/Scripts/DiscordController.cs:34)
DiscordUpdater.Update () (at Assets/Scripts/Components/DiscordUpdater.cs:7)
```

**Comment by rogeraabbccdd** [comment]
All problems should be fixed.
And for the merge conflict, I don't know should I delete comment in csv or not.

**Comment by macmillan333** [comment]
LGTM, thank you!

**Pull request merged**